extensions/transform/org.eclipse.smarthome.transform.jsonpath/src/main/java/org/eclipse/smarthome/transform/jsonpath/internal/JSonPathTransformationService.java

```diff
@@ -64,4 +72,14 @@ public String transform(String jsonPathExpression, String source) throws Transfo
 
     }
 
+    private String flattenList(List<?> list) {
+        if (list.size() == 1) {
+            return list.get(0).toString();
+        }
+        if (list.size() > 1) {
+            logger.warn("The JsonPath expression yielded more than one result: {}", list);
```

Comment:
Imho this error doesn't imply that NULL is returned...

Reply:
> What's the end user supposed to do here?

Check out the code from github and try to understand the program's behavior? 😉 

How about: 

```
Returning NULL because the JsonPath expression yielded more than one result: {} - did you forget to add a filter expression?
```

Reply:
A statement by a true developer 😁 How about

> JsonPath expression with more than one result not allowed, please adapt your selector. Result: {}